springfox-swagger-common/src/main/java/springfox/documentation/swagger/schema/ApiModelBuilder.java

```diff
@@ -48,6 +56,19 @@ public void apply(ModelContext context) {
     ApiModel annotation = AnnotationUtils.findAnnotation(forClass(context), ApiModel.class);
     if (annotation != null) {
       context.getBuilder().description(annotation.description());
+      if(StringUtils.hasText(annotation.discriminator())){
+        context.getBuilder().discriminator(annotation.discriminator());
+      }
+
+      // default parent is Void
+      if (!annotation.parent().isAssignableFrom(Void.class)) {
+        Optional<Model> parentModel = modelProvider.modelFor(ModelContext.fromParent(context, this.typeResolver.resolve(annotation.parent())));
+
+        // parent model exist because @ApiModel not required on model
+        // if(parentModel.isPresent()) {
+          context.getBuilder().parent(parentModel.get());
```

Comment:
Like I said above this could just be a `ModelRef` and eliminate the need to resolve the model in like 66